docs/quickstarts/tanstack-start.mdx

```diff
@@ -0,0 +1,243 @@
+  })(defaultStreamHandler)
+
+  export default createClerkHandler(
+    createStartHandler({
+      createRouter,
+      getRouterManifest,
+    }),
+  )(defaultStreamHandler)
+  ```
+
+  ### Add `<ClerkProvider>` and components to your app
+
+  All Clerk hooks and components must be children of the [`<ClerkProvider>`](/docs/components/clerk-provider) component, which provides active session and user context.
+
+  ```tsx {{ filename: 'app/routes/__root.tsx', ins: [11, 28, 32] }}
+  import { Outlet, ScrollRestoration, createRootRoute } from '@tanstack/react-router'
+  import { TanStackRouterDevtools } from '@tanstack/router-devtools'
+  import { Body, Head, Html, Meta, Scripts } from '@tanstack/start'
+  import * as React from 'react'
+  import { DefaultCatchBoundary } from '~/components/DefaultCatchBoundary'
+  import { NotFound } from '~/components/NotFound'
+  import { ClerkProvider } from '@clerk/tanstack-start'
+  import appCss from '~/styles/app.css?url'
+
+  export const Route = createRootRoute({
+    links: () => [{ rel: 'stylesheet', href: appCss }],
+    errorComponent: (props) => {
+      return (
+        <RootDocument>
+          <DefaultCatchBoundary {...props} />
+        </RootDocument>
+      )
+    },
+    notFoundComponent: () => <NotFound />,
+    component: () => {
+      return (
+        <ClerkProvider>
+          <RootDocument>
+            <Outlet />
+          </RootDocument>
+        </ClerkProvider>
+      )
+    },
+  })
+
+  function RootDocument({ children }: { children: React.ReactNode }) {
+    return (
+      <Html>
+        <Head>
+          <Meta />
+        </Head>
+        <Body>
+          {children}
+          <ScrollRestoration />
+          <TanStackRouterDevtools position="bottom-right" />
+          <Scripts />
+        </Body>
+      </Html>
+    )
+  }
+  ```
+
+  ### Protect your pages
+
+  #### Client side
+
+  Clerk offers [Control Components](/docs/components/overview) that allow you to protect your pages. These components are used to control the visibility of your pages based on the user's authentication state.
+
+  ```tsx {{ filename: 'app/routes/index.tsx' }}
+  import {
+    SignedIn,
+    UserButton,
+    SignOutButton,
+    SignedOut,
+    SignInButton,
+    SignUpButton,
+  } from '@clerk/tanstack-start'
+  import { createFileRoute } from '@tanstack/react-router'
+
+  export const Route = createFileRoute('/')({
+    component: Home,
+  })
+
+  function Home() {
+    return (
+      <div>
+        <h1>Index Route</h1>
+        <SignedIn>
+          <p>You are signed in!</p>
+          <div>
+            <p>View your profile here</p>
+            <UserButton />
+          </div>
+          <div>
+            <SignOutButton />
+          </div>
+        </SignedIn>
+        <SignedOut>
+          <p>You are signed out</p>
+
+          <SignInButton mode="modal" />
+
+          <SignUpButton mode="modal" />
+        </SignedOut>
+      </div>
+    )
+  }
+  ```
+
+  #### Server side
```

Comment:
Getting an error here for `to` saying that it only accepts `/` `.` and `..` as its value ?

<img width="1157" alt="Screenshot 2024-09-06 at 13 33 08" src="https://github.com/user-attachments/assets/7dc5a7e8-d970-4438-9bc8-f1ca2ec1cfcc">

Reply:
This is expected as the route does not exists and everything in TanStack Start is fully typed, this is there as placeholder